src/test/java/teammates/test/cases/ui/pagedata/StudentProfilePageDataTest.java

```diff
@@ -0,0 +1,80 @@
+
+public class StudentProfilePageDataTest {
+    
+    private StudentProfileAttributes spa;
+    private AccountAttributes acct;
+    private String isEditingPhoto;
+    private String pictureUrl;
+
+    private StudentProfilePageData sppd;
+
+    @Test
+    public void testAll() {
+        testWithPictureKeyAndNoNullFields();
+        testWithNoPictureKeyAndNullFields();
+    }
+
+    private void testWithPictureKeyAndNoNullFields() {
+        sppd = initializeDataWithPictureKeyAndNoNullFields();
+        testProfileEditBox(sppd.getProfileEditBox());
+        testUploadPhotoModal(sppd.getUploadPhotoModal());
+    }
+
+    private void testWithNoPictureKeyAndNullFields() {
+        sppd = initializeDataWithNoPictureKeyAndNullFields();
+        testProfileEditBox(sppd.getProfileEditBox());
+        testUploadPhotoModal(sppd.getUploadPhotoModal());
+    }
+
+    private StudentProfilePageData initializeDataWithPictureKeyAndNoNullFields() {
+        spa = new StudentProfileAttributes("valid.id.2", "short name", "e@mail2.com", "inst", "nationality",
+                                           "male", "more info", "pictureKey");
+        acct = new AccountAttributes("valid.id", "full name", false, "e@mail1.com", "inst", spa);
+        isEditingPhoto = "false";
+        pictureUrl = Const.ActionURIs.STUDENT_PROFILE_PICTURE
+                   + "?" + Const.ParamsNames.BLOB_KEY + "=" + spa.pictureKey
+                   + "&" + Const.ParamsNames.USER_ID + "=" + acct.googleId;
+        return new StudentProfilePageData(acct, isEditingPhoto);        
+    }
+    
+    private StudentProfilePageData initializeDataWithNoPictureKeyAndNullFields() {
+        spa = new StudentProfileAttributes("valid.id.2", null, null, null, null, "male", null, "");
+        acct = new AccountAttributes("valid.id", "full name", false, "e@mail1.com", "inst", spa);
+        pictureUrl = Const.SystemParams.DEFAULT_PROFILE_PICTURE_PATH;
+        return new StudentProfilePageData(acct, isEditingPhoto);        
+    }
+    
+    private void testProfileEditBox(StudentProfileEditBox profileEditBox) {
+        assertEquals(acct.name, profileEditBox.getName());
+        assertEquals(isEditingPhoto, profileEditBox.getEditingPhoto());
+        assertEquals(StringHelper.convertToEmptyStringIfNull(spa.shortName), profileEditBox.getShortName());
+        // email comes from SPA, not AA
+        assertEquals(StringHelper.convertToEmptyStringIfNull(spa.email), profileEditBox.getEmail());
+        assertEquals(StringHelper.convertToEmptyStringIfNull(spa.institute), profileEditBox.getInstitute());
+        assertEquals(StringHelper.convertToEmptyStringIfNull(spa.nationality), profileEditBox.getNationality());
+        assertEquals(spa.gender, profileEditBox.getGender());
+        assertEquals(StringHelper.convertToEmptyStringIfNull(spa.moreInfo), profileEditBox.getMoreInfo());
+        assertEquals(acct.googleId, profileEditBox.getGoogleId()); // googleId comes from AA, not SPA
```

Comment:
"why" comments would probably help more than "what" here.
Why does email and googleId specifically come from SPA and AA respectively when they (variables named the same) appear to be available in both?